skfp/preprocessing/filters/rule_of_2.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumHBA, CalcNumHBD
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class RuleOf2(BaseFilter):
+    """
+    Rule of 2 defined  in [1].
+
+    Proposed rule for finding reagent to building block design.
+
+    The proposed reagents according to this formula should meet the following criteria:
+    - molecular weight <= 200 daltons
+    - HBA <= 4
+    - HBD <= 2
+    - logP <= 2
+
+    Parameters
+    ----------
+    allow_one_violation : bool, default=True
+        Whether to allow violating one of the rules for a molecule. This makes the
+        filter less restrictive, and is the part of the original definition of this
+        filter.
+
+    n_jobs : int, default=None
+        The number of jobs to run in parallel. :meth:`transform_x_y` and
+        :meth:`transform` are parallelized over the input molecules. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors. See Scikit-learn documentation on ``n_jobs`` for more details.
+
+    batch_size : int, default=None
+        Number of inputs processed in each batch. ``None`` divides input data into
+        equal-sized parts, as many as ``n_jobs``.
+
+    verbose : int, default=0
+        Controls the verbosity when generating conformers.
+
+    References
+    -----------
+    .. [1] 'Goldberg, F. W., Kettle, J. G., Kogej, T., Perry, M. W. D., & Tomkinson, N. P. (2015).
+        Designing novel building blocks is an overlooked strategy to improve compound quality.
+        Drug Discovery Today, 20(1), 11–17. doi:10.1016/j.drudis.2014.09.023'_
```

Comment:
1. Those should be backticks ` not apostrophes ' to create proper link
2. Keep formatting of other literature references. In separate lines: list of authors, title in quotes, journal notes, link in <>